core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -653,6 +657,28 @@ Cuda::size_type *CudaInternal::scratch_functor(
   return m_scratchFunctor;
 }
 
+void *CudaInternal::resize_team_scratch_space(std::int64_t bytes,
+                                              bool force_shrink) {
+  if (m_team_scratch_current_size == 0) {
+    m_team_scratch_current_size = bytes;
+    m_team_scratch_ptr          = Kokkos::kokkos_malloc<Kokkos::CudaSpace>(
+        "CudaSpace::ScratchMemory", m_team_scratch_current_size);
+  }
+  if (bytes > m_team_scratch_current_size) {
+    m_team_scratch_current_size = bytes;
+    Kokkos::kokkos_free<Kokkos::CudaSpace>(m_team_scratch_ptr);
+    m_team_scratch_ptr = Kokkos::kokkos_malloc<Kokkos::CudaSpace>(
+        "CudaSpace::ScratchMemory", m_team_scratch_current_size);
```

Comment:
Why did you chose this over `kokkos_realloc`?

Reply:
yeah we should do kokkos_realloc